base/workflow/inst/batch_runs.R

```diff
@@ -0,0 +1,431 @@
+  filter(container_type == "Model") %>%
+  left_join(tbl(bety, "models"), c("container_id" = "id")) %>%
+  select(model_id = container_id, model_name, revision,
+         file_name, file_path, dbfile_id = id, ) %>%
+  collect() %>%
+  mutate(exists = file.exists(file.path(file_path, file_name)))
+
+message("Found the following models on the machine:")
+print(model_df)
+
+if (!all(model_df$exists)) {
+  message("WARNING: The following models are registered on the machine ",
+          "but their files do not exist:")
+  model_df %>%
+    filter(!exists) %>%
+    print()
+
+  model_df <- model_df %>%
+    filter(exists)
+}
+
+models <- model_df[["model_id"]]
+
+## Find Sites
+## Site with no inputs from any machines that is part of Ameriflux site group and Fluxnet Site group
+site_id_noinput <- tbl(bety, "sites") %>%
+  anti_join(tbl(bety, "inputs")) %>%
+  inner_join(tbl(bety, "sitegroups_sites") %>%
+               filter(sitegroup_id == 1),
+             by = c("id" = "site_id")) %>%
+  dplyr::select("id.x", "notes", "sitename") %>%
+  dplyr::filter(grepl("TOWER_BEGAN", notes)) %>%
+  collect() %>%
+  dplyr::mutate(
+    # Grab years from string within the notes
+    start_year = substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_BEGAN = ).*(?=  TOWER_END)")),1,4),
+    #Empty tower end in the notes means that it goes until present day so if empty enter curent year.
+    end_year = dplyr::if_else(
+      substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_END = ).*(?=)")),1,4) == "",
+      as.character(lubridate::year(Sys.Date())),
+      substring(stringr::str_extract(notes,pattern = ("(?<=TOWER_END = ).*(?=)")),1,4)
+    ),
+    #Check if startdate year is within the inerval of that is given
+    in_date = data.table::between(as.numeric(lubridate::year(startdate)),as.numeric(start_year),as.numeric(end_year))
+  ) %>%
+  dplyr::filter(
+    in_date,
+    as.numeric(end_year) - as.numeric(start_year) > 1
+  ) %>%
+  mutate(sitename = gsub(" ", "_", sitename)) %>%
+  rename(site_id = id.x)
+
+message("Running tests at ", nrow(site_id_noinput), " sites:")
+print(site_id_noinput)
+
+site_id <- site_id_noinput$site_id
+site_name <- gsub(" ", "_", site_id_noinput$sitename)
+
+
+
+#Create permutations of arg combinations
+options(scipen = 999)
+run_table <- expand.grid(
+  model_id = models,
+  met = met_name,
+  site_id = site_id,
+  start_date = startdate,
+  end_date = enddate,
+  pecan_path = pecan_path,
+  ensemble_variable = out.var,
+  # ensemble = ensemble,
+  ensemble_size = ens_size,
+  sensitivity = sensitivity,
+  user_id = user_id,
+  dbfiles_folder = dbfiles_folder,
+  stringsAsFactors = FALSE
+)
+#Execute function to spit out a table with a column of NA or success
+
+result_table <- as_tibble(run_table) %>%
+  mutate(
+    outdir = NA_character_,
+    workflow_complete = NA,
+    has_jobsh = NA,
+    model_output_raw = NA,
+    model_output_processed = NA
+  )
+
+for (i in seq_len(nrow(run_table))) {
```

Comment:
probably fine for a first pass, but in general I don't think we should be limited to doing these runs sequentially

Reply:
Moved to new issue #2399